File: server/src/main/java/com/tomato/bookstore/model/Order.java
package com.tomato.bookstore.model;

import com.tomato.bookstore.constant.OrderStatus;
import com.tomato.bookstore.constant.PaymentMethod;
import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/** 订单实体类 */
@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "orders")
public class Order {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id", nullable = false)
  private User user;

  @Column(nullable = false)
  private BigDecimal totalAmount;

  @Column(nullable = false)
  @Enumerated(EnumType.STRING)
  private PaymentMethod paymentMethod;

  @Column(nullable = false)
  @Enumerated(EnumType.STRING)
  private OrderStatus status;

  @Column(nullable = false)
  private String shippingAddress;

  private String tradeNo;

  private LocalDateTime paymentTime;

  @Column(nullable = false)
  private LocalDateTime createdAt;

  private LocalDateTime updatedAt;

  @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
  @Builder.Default
  private List<CartsOrdersRelation> orderItems = new ArrayList<>();
}

File: client/src/types/order.ts
import type { Cart } from "./cart";

/**
 * 支付方式枚举
 */
export enum PaymentMethod {
  ALIPAY = "ALIPAY",
}

/**
 * 订单状态枚举
 */
export enum OrderStatus {
  PENDING = "PENDING", // 待支付
  PAID = "PAID", // 已支付
  CANCELLED = "CANCELLED", // 已取消
  TIMEOUT = "TIMEOUT", // 已超时
}

/**
 * 订单信息
 */
export interface Order {
  orderId?: number;
  userId?: number;
  totalAmount: number;
  paymentMethod: PaymentMethod;
  status: OrderStatus;
  shippingAddress: string;
  tradeNo?: string;
  paymentTime?: string;
  createTime?: string;
  updateTime?: string;
  orderItems: Cart[];
}

/**
 * 支付信息
 */
export interface Payment {
  paymentForm: string;
  orderId: number;
  totalAmount: number;
  paymentMethod: PaymentMethod;
}
